samples/99_zil_interpreter/app/builtins.rb

```diff
@@ -262,3 +262,121 @@ def expect_minimum_argument_count!(arguments, min_count)
   false
 }
 
+# Tables are represented internally as arrays which contain "bytes" which can be set/get with PUTB/GETB
+# Every other data type is saved as a "word" so basically 2 bytes -
+# but for convenience's sake those words are just stored as two byte elements: [value, 0]
+# Since in Zork single bytes of words are never accessed - nor vice-versa - this kind of representation
+# should be fine.
+ZIL_BUILTINS[:TABLE] = define_for_evaled_arguments { |arguments|
+  flags, values = ZIL::Table.get_flags_and_values arguments
+
+  ZIL::Table.build flags: flags, values: values
+}
+
+ZIL_BUILTINS[:ITABLE] = define_for_evaled_arguments { |arguments|
+  size = arguments[0]
+  flags, default_values = ZIL::Table.get_flags_and_values arguments[1..-1]
+
+  ZIL::Table.build flags: flags, values: default_values * size
+}
+
+ZIL_BUILTINS[:LTABLE] = define_for_evaled_arguments { |arguments|
+  flags, values = ZIL::Table.get_flags_and_values arguments
+  flags << :LENGTH unless flags.include? :LENGTH
+
+  ZIL::Table.build flags: flags, values: values
+}
```

Comment:
I think that these should also check for minimum argument count?

Reply:
Right `ITABLE` at least needs to check for a minimum argument count - good catch!
The other two can technically produce empty tables I think so I don't think they need count validation

Reply:
Ah and I just realized I forgot to implement a feature of ITABLE 😅 that is used in Zork1 (First argument being a type specifier and not the length) So I'll add that,too